src/dotnet/APIView/APIView/NamedTypeAPIV.cs

```diff
@@ -0,0 +1,142 @@
+using System.Text;
+
+namespace APIView
+{
+    public class NamedTypeAPIV
+    {
+        public string Name { get; }
+        public string Type { get; }
+        public string EnumUnderlyingType { get; }
+
+        public ImmutableArray<EventAPIV> Events { get; }
+        public ImmutableArray<FieldAPIV> Fields { get; }
+        public ImmutableArray<string> Implementations { get; }
+        public ImmutableArray<MethodAPIV> Methods { get; }
+        public ImmutableArray<NamedTypeAPIV> NamedTypes { get; }
+        public ImmutableArray<PropertyAPIV> Properties { get; }
+
+        /// <summary>
+        /// Construct a new namedType instance, represented by the provided symbol.
+        /// </summary>
+        /// <param name="symbol">The symbol representing the named type.</param>
+        public NamedTypeAPIV(INamedTypeSymbol symbol)
+        {
+            this.Name = symbol.Name;
+            this.Type = symbol.TypeKind.ToString().ToLower();
+            if (symbol.EnumUnderlyingType != null)
+                this.EnumUnderlyingType = symbol.EnumUnderlyingType.ToDisplayString();
+
+            List<EventAPIV> events = new List<EventAPIV>();
+            List<FieldAPIV> fields = new List<FieldAPIV>();
+            List<string> implementations = new List<string>();
+            List<MethodAPIV> methods = new List<MethodAPIV>();
+            List<NamedTypeAPIV> namedTypes = new List<NamedTypeAPIV>();
+            List<PropertyAPIV> properties = new List<PropertyAPIV>();
+
+            // add any types declared in the body of this type to lists
+            foreach (var memberSymbol in symbol.GetMembers())
+            {
+                if (memberSymbol.DeclaredAccessibility != Accessibility.Public) continue;
+
+                switch (memberSymbol)
+                {
+                    case IEventSymbol e:
+                        events.Add(new EventAPIV(e));
+                        break;
+
+                    case IFieldSymbol f:
+                        fields.Add(new FieldAPIV(f));
+                        break;
+
+                    case IMethodSymbol m:
+                        bool autoMethod = false;
+                        if (m.AssociatedSymbol != null)
+                            autoMethod = (m.AssociatedSymbol.Kind == SymbolKind.Event) || (m.AssociatedSymbol.Kind == SymbolKind.Property);
+
+                        if (!(m.Name.Equals(".ctor") || autoMethod))
+                            methods.Add(new MethodAPIV(m));
+                        break;
+
+                    case INamedTypeSymbol n:
+                        namedTypes.Add(new NamedTypeAPIV(n));
+                        break;
+
+                    case IPropertySymbol p:
+                        properties.Add(new PropertyAPIV(p));
+                        break;
+                }
+            }
+
+            // add a string representation of each implemented type to list
+            foreach (var i in symbol.Interfaces)
```

Comment:
Doesn't ToDisplayString work here?

Reply:
It would include the namespace of the interface, but would definitely be easier (and more explicit). Just changed it!